codecs/avif/enc/avif_enc.cpp

```diff
@@ -32,18 +35,33 @@ val encode(std::string buffer, int width, int height, AvifOptions options) {
   avifPixelFormat format;
   switch (options.subsample) {
     case 0:
-      format = AVIF_PIXEL_FORMAT_YUV420;
+      format = AVIF_PIXEL_FORMAT_YUV400;
       break;
     case 1:
-      format = AVIF_PIXEL_FORMAT_YUV422;
+      format = AVIF_PIXEL_FORMAT_YUV420;
       break;
     case 2:
+      format = AVIF_PIXEL_FORMAT_YUV422;
+      break;
+    case 3:
       format = AVIF_PIXEL_FORMAT_YUV444;
       break;
   }
 
   avifImage* image = avifImageCreate(width, height, depth, format);
 
+  if (
+    options.maxQuantizer == AVIF_QUANTIZER_LOSSLESS &&
+    options.minQuantizer == AVIF_QUANTIZER_LOSSLESS &&
+    options.minQuantizerAlpha == AVIF_QUANTIZER_LOSSLESS &&
+    options.maxQuantizerAlpha == AVIF_QUANTIZER_LOSSLESS &&
+    format == AVIF_PIXEL_FORMAT_YUV444
+  ) {
+    image->matrixCoefficients = AVIF_MATRIX_COEFFICIENTS_IDENTITY;
+  } else {
+    image->matrixCoefficients = AVIF_MATRIX_COEFFICIENTS_BT709;
```

Comment:
Chrome's decoder gets the colours wrong unless we do this.

Reply:
BT709 is compatible with sRGB. Anything else is asking for trouble.